File: Java/1991. Find the Middle Index in Array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int l = 0, r = 0, t;

        for(int i: nums) r += i;

        for(int i = 0; i < nums.length; i++){
            t = nums[i];
            if(l == r - t) return i;
            l += t;
            r -= t;
        }

        return -1;
    }
}

File: Python/1991. Find the Middle Index in Array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        l, r = 0, sum(nums)

        for i in range(len(nums)):
            if l == r - nums[i]: return i

            l += nums[i]
            r -= nums[i]
        
        return -1
